build-tools/timing/timing.targets

```diff
@@ -63,15 +63,42 @@
     </PropertyGroup>
     <!--Simulate a clean checkout with git commands-->
     <Git
-        Arguments="rm -r --cached %(XACaptureBuildTimingProject.DirectoryToClean)"
+        Condition=" '%(XACaptureBuildTimingProject.Repo)' == '' "
+        Arguments="rm -r --cached %(XACaptureBuildTimingProject.DirectoryFullPath)"
         WorkingDirectory="$(_TopDir)"
         ToolPath="$(GitToolPath)"
-        ToolExe="$(GitToolExe)" />
+        ToolExe="$(GitToolExe)"
+    />
+    <Git
+        Condition=" '%(XACaptureBuildTimingProject.Repo)' == '' "
+        Arguments="checkout HEAD %(XACaptureBuildTimingProject.DirectoryFullPath)"
+        WorkingDirectory="$(_TopDir)"
+        ToolPath="$(GitToolPath)"
+        ToolExe="$(GitToolExe)"
+    />
+    <!--If the XACaptureBuildTimingProject defines Repo, we need to clone it-->
+    <RemoveDir
+        Condition=" '%(XACaptureBuildTimingProject.Repo)' != '' And Exists('%(XACaptureBuildTimingProject.DirectoryFullPath)')"
+        Directories="%(XACaptureBuildTimingProject.DirectoryFullPath)"
+    />
     <Git
-        Arguments="checkout HEAD %(XACaptureBuildTimingProject.DirectoryToClean)"
+        Condition=" '%(XACaptureBuildTimingProject.Repo)' != '' "
+        Arguments="clone &quot;%(XACaptureBuildTimingProject.Repo)&quot; --depth=1 &quot;%(XACaptureBuildTimingProject.DirectoryFullPath)&quot;"
         WorkingDirectory="$(_TopDir)"
         ToolPath="$(GitToolPath)"
-        ToolExe="$(GitToolExe)" />
+        ToolExe="$(GitToolExe)"
+    />
+    <Git
+        Condition=" '%(XACaptureBuildTimingProject.Repo)' != '' And '%(XACaptureBuildTimingProject.Commit)' != '' "
+        Arguments="checkout &quot;%(XACaptureBuildTimingProject.Commit)&quot;"
+        WorkingDirectory="%(XACaptureBuildTimingProject.DirectoryFullPath)"
+        ToolPath="$(GitToolPath)"
+        ToolExe="$(GitToolExe)"
+    />
+    <Exec 
+        Condition=" '%(XACaptureBuildTimingProject.Restore)' != '' "
+        Command="$(_TopDir).nuget\NuGet.exe restore %(XACaptureBuildTimingProject.Restore)"
```

Comment:
What is the `%(Restore)` metadata, and should it be quoted?

Reply:
It's in the XML example above, you need a path to an SLN, which we don't have anywhere else.

It _does_ need to be quoted, I'll fix that.